Feature/add tox configs tests and fixes

otahontas opened the pull request:
### Description

(Rebased to state where previous PR was merged to develop).

This PR adds mypy and pydocstyle to tox and adds a couple of following smaller fixes:
- fixes some mypy-errors and adds missing type hints
- adds some try-catch checks for database and HTTP errors
- adds accession number generation
- makes it possible to submit multiple file same schema in one submission
- adds receipt generation
- adds some better submission.xml parsing

As discussed, receipt generator and submission.xml won't probably be used that much, and focus will be moved on to more rest-compliant endpoint features. Yet, they'll be now present in code base in case they're needed at some point (maybe for compatibility with EBI's service or what not). 

Tests are also now written in a way that doesn't result any async code running synchronously. Tests can be also now ran before each-commit with included pre-commit -hook. I'll add documentation for this soon.

### Related issues
Fixes #7 

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update (following up in this PR)

### Changes Made

See git commit log below.

### Testing

- [x] Unit Tests

blankdots commented on tox.ini:
```diff
@@ -21,6 +21,22 @@ deps =
+[testenv:pydocstyle]
+skip_install = true
+deps =
+    pydocstyle
```
if you add https://pypi.org/project/flake8-docstrings/ as a dependency at the `testev:flake8` then this does not need to be done in a separate env.

blankdots replied:
we will have a separate stage for generating and testing documentation, so this could be included with flake8

blankdots commented on tox.ini:
```diff
@@ -1,13 +1,13 @@
 [tox]
-envlist = py{36,37},flake8,isort
+envlist = py{36,37},flake8,isort,mypy,pydocstyle
```
should we add these to travis integration as well ?

otahontas commented:
@blankdots I changed some code based on you comments. I also fixed bunch of typos from docstring (didn't notice them before), no effect on code itself. 

This should be now ok to merge.
